docs/Makefile

```diff
@@ -3,7 +3,7 @@
 
 # You can set these variables from the command line, and also
 # from the environment for the first two.
-SPHINXOPTS    ?= -W --keep-going
+SPHINXOPTS    ?= -W --keep-going -Q
```

Comment:
`-Q` would also suppress warnings, which we consider errors (see `-W` flag). In my testing on this branch, if I introduce a warning, I get:
```
$ make html
/Users/hanlintang/composer-fork/docs/source/conf.py:108: UserWarning: The git repo is dirty. The commit sha for source code links will be incorrect.
  warnings.warn(warning_msg)
Generating rst file /Users/hanlintang/composer-fork/docs/source/api_reference/composer.rst for module: composer
Generating rst file /Users/hanlintang/composer-fork/docs/source/api_reference/composer.algorithms.rst for module: composer.algorithms
Generating rst file /Users/hanlintang/composer-fork/docs/source/api_reference/composer.callbacks.rst for module: composer.callbacks
Generating rst file /Users/hanlintang/composer-fork/docs/source/api_reference/composer.core.rst for module: composer.core
Generating rst file /Users/hanlintang/composer-fork/docs/source/api_reference/composer.core.types.rst for module: composer.core.types
Generating rst file /Users/hanlintang/composer-fork/docs/source/api_reference/composer.datasets.rst for module: composer.datasets
Generating rst file /Users/hanlintang/composer-fork/docs/source/api_reference/composer.functional.rst for module: composer.functional
Generating rst file /Users/hanlintang/composer-fork/docs/source/api_reference/composer.functional.rst for module: composer.functional
Generating rst file /Users/hanlintang/composer-fork/docs/source/api_reference/composer.loggers.rst for module: composer.loggers
Generating rst file /Users/hanlintang/composer-fork/docs/source/api_reference/composer.loss.rst for module: composer.loss
Generating rst file /Users/hanlintang/composer-fork/docs/source/api_reference/composer.metrics.rst for module: composer.metrics
Generating rst file /Users/hanlintang/composer-fork/docs/source/api_reference/composer.models.rst for module: composer.models
[ommited: several "generating.. lines]
make: *** [html] Error 1
```

which is missing the critical error:
```
/Users/hanlintang/composer-fork/docs/source/trainer/using_the_trainer.rst:462: WARNING: unknown document: /abc
```

Could we revert the `-Q`?

Reply:
Oh good call. It actually should be `-q`. See https://www.sphinx-doc.org/en/master/man/sphinx-build.html#cmdoption-sphinx-build-q

Reply:
With `-q`, we will get the proper warning:
```
(ec-venv) Evans-MacBook-Pro:docs evanracah$ make html
/Users/evanracah/code/evan-composer/docs/source/conf.py:108: UserWarning: The git repo is dirty. The commit sha for source code links will be incorrect.
  warnings.warn(warning_msg)
Generating rst file /Users/evanracah/code/evan-composer/docs/source/api_reference/composer.rst for module: composer
Generating rst file /Users/evanracah/code/evan-composer/docs/source/api_reference/composer.algorithms.rst for module: composer.algorithms
Generating rst file /Users/evanracah/code/evan-composer/docs/source/api_reference/composer.callbacks.rst for module: composer.callbacks
Generating rst file /Users/evanracah/code/evan-composer/docs/source/api_reference/composer.core.rst for module: composer.core
Generating rst file /Users/evanracah/code/evan-composer/docs/source/api_reference/composer.core.types.rst for module: composer.core.types
Generating rst file /Users/evanracah/code/evan-composer/docs/source/api_reference/composer.datasets.rst for module: composer.datasets
Generating rst file /Users/evanracah/code/evan-composer/docs/source/api_reference/composer.functional.rst for module: composer.functional
Generating rst file /Users/evanracah/code/evan-composer/docs/source/api_reference/composer.functional.rst for module: composer.functional
Generating rst file /Users/evanracah/code/evan-composer/docs/source/api_reference/composer.loggers.rst for module: composer.loggers
Generating rst file /Users/evanracah/code/evan-composer/docs/source/api_reference/composer.loss.rst for module: composer.loss
Generating rst file /Users/evanracah/code/evan-composer/docs/source/api_reference/composer.metrics.rst for module: composer.metrics
Generating rst file /Users/evanracah/code/evan-composer/docs/source/api_reference/composer.models.rst for module: composer.models
Generating rst file /Users/evanracah/code/evan-composer/docs/source/api_reference/composer.optim.rst for module: composer.optim
Generating rst file /Users/evanracah/code/evan-composer/docs/source/api_reference/composer.profiler.rst for module: composer.profiler
Generating rst file /Users/evanracah/code/evan-composer/docs/source/api_reference/composer.utils.rst for module: composer.utils
Generating rst file /Users/evanracah/code/evan-composer/docs/source/api_reference/composer.utils.dist.rst for module: composer.utils.dist
Generating rst file /Users/evanracah/code/evan-composer/docs/source/api_reference/composer.utils.reproducibility.rst for module: composer.utils.reproducibility
/Users/evanracah/code/evan-composer/docs/source/trainer/using_the_trainer.rst:462: WARNING: unknown document: /abc
make: *** [html] Error 1
```

Reply:
fixed

Reply:
Ok I added `-q` just to `make doctest`, so `make html` can still be a little bit verbose.